x-pack/plugins/ml/public/jobs/new_job/simple/components/enable_model_plot_checkbox/enable_model_plot_checkbox_directive.js

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import { EnableModelPlotCheckbox } from './enable_model_plot_checkbox_view.js';
+import { ml } from 'plugins/ml/services/ml_api_service';
+
+import { uiModules } from 'ui/modules';
+const module = uiModules.get('apps/ml');
+
+module.directive('mlEnableModelPlotCheckbox', function () {
+  return {
+    restrict: 'AE',
+    replace: false,
+    scope: {
+      formConfig: '=',
+      ui: '=ui',
+      getJobFromConfig: '='
+    },
+    link: function ($scope, $element) {
+      const STATUS = {
+        FAILED: -1,
+        NOT_RUNNING: 0,
+        RUNNING: 1,
+        FINISHED: 2,
+        WARNING: 3,
+      };
+      const errorHandler = (error) => {
+        console.log('Cardinality could not be validated', error);
+        $scope.ui.cardinalityValidator.status = STATUS.FAILED;
+        $scope.ui.cardinalityValidator.message = 'Cardinality could not be validated';
+      };
+
+      // Only model plot cardinality relevant
+      // format:[{id:"cardinality_model_plot_high",modelPlotCardinality:11405}, {id:"cardinality_partition_field",fieldName:"clientip"}]
+      const checkCardinalitySuccess = (data) => {
+        const response = {
+          success: true,
+        };
+        // There were no fields to run cardinality on.
+        if (Array.isArray(data) && data.length === 0) {
+          return response;
+        }
+
+        for (let i = 0; i < data.length; i++) {
+          if (data[i].id === 'success_cardinality') {
+            break;
+          }
+
+          if (data[i].id === 'cardinality_model_plot_high') {
+            response.success = false;
+            response.highCardinality = data[i].modelPlotCardinality;
+            break;
+          }
+        }
+
+        return response;
+      };
+
+      const validateCardinality = function () {
+        $scope.ui.cardinalityValidator.status = STATUS.RUNNING;
+        $scope.ui.cardinalityValidator.message = '';
+
+        // create temporary job since cardinality validation expects that format
+        const tempJob = $scope.getJobFromConfig($scope.formConfig);
+
+        ml.validateCardinality(tempJob)
+          .then((response) => {
+            const validationResult = checkCardinalitySuccess(response);
+
+            if (validationResult.success === true) {
+              $scope.formConfig.enableModelPlot = true;
+              $scope.ui.cardinalityValidator.status = STATUS.FINISHED;
+            } else {
+              $scope.ui.cardinalityValidator.message = `The estimated cardinality of ${validationResult.highCardinality}
+                of fields relevant to creating model plots might result in resource intensive jobs.`;
```

Comment:
Only one job is getting created here, so change the end part of the message to 'in a resource intensive job'. Does the response. Is there any way we could use the same String as the `job_validation` code on the server? That might be best left for the localization work?